ipld/amt/src/iter.rs

```diff
@@ -41,17 +42,110 @@ where
     /// # anyhow::Ok(())
     /// ```
     pub fn iter(&self) -> Iter<'_, V, &BS, Ver> {
-        Iter {
-            stack: vec![IterStack {
-                node: &self.root.node,
-                idx: 0,
-            }],
-            height: self.root.height,
+        Iter::new(
+            &self.root.node,
+            &self.block_store,
+            self.height(),
+            self.bit_width(),
+            0,
+        )
+    }
+
+    /// Iterate over the AMT from the given starting point.
+    ///
+    /// ```rust
+    /// use fvm_ipld_amt::Amt;
+    /// use fvm_ipld_blockstore::MemoryBlockstore;
+    ///
+    /// let store = MemoryBlockstore::default();
+    ///
+    /// let mut amt = Amt::new(store);
+    /// let kvs: Vec<u64> = (0..=5).collect();
+    /// kvs
+    ///     .iter()
+    ///     .map(|k| amt.set(u64::try_from(*k).unwrap(), k.to_string()))
```

Comment:
```suggestion
    ///     .map(|k| amt.set(*k, k.to_string()))
```
?

Reply:
Hm. well, I'd need to change the line above to `0u64..=5` (I'm copying this document from the existing method, but you're right that that way is better).

Maybe file an issue and I can fix both `iter` and `iter_from` in another PR?

Reply:
https://github.com/filecoin-project/ref-fvm/pull/1971 there you go, I don't think you need `0u64..=5` cause the type is specified on the receiver